nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/status/history/ProcessGroupStatusDescriptor.java

```diff
@@ -113,16 +113,16 @@ public String getField() {
 
 
     private static long calculateTaskMillis(final ProcessGroupStatus status) {
-        long nanos = 0L;
```

Comment:
I think I mentioned this at the PR description, but the point would be to avoid loss of information: with the original code, at every level of the recursion we did a nano > millis conversion, but the caller (one level up in the recursion) would consider the result as nano. Thus, the deeper we are in the group structure, the more times we make a conversion, which looks to be incorrect.

Reply:
Something like this would solve both issues:

```
private static long calculateTaskMillis(final ProcessGroupStatus status) {
    return TimeUnit.MILLISECONDS.convert(calculateTaskNanos(status), TimeUnit.NANOSECONDS);
}

private static long calculateTaskNanos(final ProcessGroupStatus status) {
   long nanos = 0L;

   for (final ProcessorStatus procStatus : status.getProcessorStatus()) {
       nanos += procStatus.getProcessingNanos();
   }

   for (final ProcessGroupStatus childStatus : status.getProcessGroupStatus()) {
       nanos += calculateTaskNanos(childStatus);
   }

   return nanos;
}
```

Reply:
Ah, I see. Yes, I think this is a good approach, to calculate recursively using nanos and then converting to millis only after the recursive call.